src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/health/health.component.html

```diff
@@ -88,151 +97,157 @@
-                  class="cd-col-5"
-                  cardClass="card-medium"
-                  contentClass="content-medium content-highlight"
-                  *ngIf="contentData.df?.stats?.total_objects">
-      {{ contentData.df?.stats?.total_objects }}
-    </cd-info-card>
+      <cd-info-card cardTitle="Objects"
+                    i18n-cardTitle
+                    class="cd-col-5"
+                    cardClass="card-medium"
+                    contentClass="content-medium content-highlight"
+                    *ngIf="contentData.df?.stats?.total_objects !== undefined">
+        {{ contentData.df?.stats?.total_objects }}
+      </cd-info-card>
 
-    <cd-info-card cardTitle="PGs per OSD"
-                  i18n-cardTitle
-                  class="cd-col-5"
-                  cardClass="card-medium"
-                  contentClass="content-medium content-highlight"
-                  *ngIf="contentData.pg_info">
-      {{ contentData.pg_info.pgs_per_osd | dimless }}
-    </cd-info-card>
+      <cd-info-card cardTitle="PGs per OSD"
+                    i18n-cardTitle
+                    class="cd-col-5"
+                    cardClass="card-medium"
+                    contentClass="content-medium content-highlight"
+                    *ngIf="contentData.pg_info">
+        {{ contentData.pg_info.pgs_per_osd | dimless }}
+      </cd-info-card>
 
-    <cd-info-card cardTitle="PG Status"
-                  i18n-cardTitle
-                  class="cd-col-5"
-                  cardClass="card-medium"
-                  contentClass="card-chart"
-                  (click)="pgStatusTarget.toggle()">
-      <ng-template #pgStatus>
-        <p class="logs-link"
-           i18n>
-          &rarr; See <a (click)="viewportScroller.scrollToAnchor('logs')">Logs</a> for more details.
-        </p>
-        <ul>
-          <li *ngFor="let pgStatesText of contentData.pg_info.statuses | keyvalue">
-            {{ pgStatesText.key }}: {{ pgStatesText.value }}
-          </li>
-        </ul>
-      </ng-template>
-      <div class="pg-status-popover-wrapper">
-        <div [popover]="pgStatus"
-             triggers=""
-             #pgStatusTarget="bs-popover"
-             placement="bottom">
-          <cd-health-pie [data]="contentData"
-                         chartType="pie"
-                         [displayLegend]="true"
-                         (prepareFn)="preparePgStatus($event[0], $event[1])">
-          </cd-health-pie>
+      <cd-info-card cardTitle="PG Status"
+                    i18n-cardTitle
+                    class="cd-col-5"
+                    cardClass="card-medium"
+                    contentClass="content-chart"
+                    (click)="pgStatusTarget.toggle()"
+                    *ngIf="contentData.pg_info">
+        <ng-template #pgStatus>
+          <p class="logs-link"
+             i18n>
+            &rarr; See <a (click)="viewportScroller.scrollToAnchor('logs')">Logs</a> for more details.
```

Comment:
Currently the JS code must be translated too. If there is a mistake in the translation, then the function will be broken or a more dangereous scenario might happen, unwanted code is executed introduced via translation files. My question is: Is it possible to use something like

```
"&rarr; See <a %s>Logs</a> for more details.".format("(click)=\"viewportScroller.scrollToAnchor('logs')\"")
```

Reply:
One solution could be to split the text in 3 elements and, in order to not lose translation context, add a description to [help the translator](https://angular.io/guide/i18n#help-the-translator-with-a-description-and-meaning). Something like this:

          <p class="logs-link">
            &rarr;
            <ng-container i18n="1st part of 'See Logs for more details.'">See</ng-container>
            <a (click)="viewportScroller.scrollToAnchor('logs')" i18n="2nd part of 'See Logs for more details.'"> Logs </a>
            <ng-container i18n="3rd part of 'See Logs for more details.'">for more details.</ng-container>
          </p>

Reply:
@Exotelis Any comments on that?

Reply:
I'd recommend leave it like it is, for now. As far as I can tell the angular community is also waiting for something like a sprintf for i18n.
It should not be a security risk as long as we review the translation changes.